### ISSUE TITLE
Breaking change in 2.1.0 due to switch from self_link to id

### ISSUE THREAD

**Issue by kpocius** [terraform-google-modules/terraform-google-kms#53]
### TL;DR

#52 makes it sound like the only change is constraints, but #50 also changes `google_kms_key_ring.key_ring.self_link` to `google_kms_key_ring.key_ring.id` which results in all `google_kms_crypto_key_iam_binding` resources needing to be re-created.

### Expected behavior

No change in terraform plan after updating the module

### Observed behavior

All `google_kms_crypto_key_iam_binding` resources are marked for replacement

### Terraform Configuration

```hcl
Create a key with minimal example, using version 2.0.1, then change version to 2.1.0
```

### Terraform Version

```sh
1.1.2
```

### Additional information

Breaking change and potential migration path should be indicated in the release notes.

**Comment by bharathkkb** [comment]
Thanks for the report @kpocius 
IIRC the values for self link and id were identical, could you provide a repro? I tried as below but did not see any diffs.
```tf
module "kms" {
  source  = "terraform-google-modules/kms/google"
  version = "2.0.1" #change to 2.1.0

  project_id         = local.project_id
  location           = "europe"
  keyring            = "sample-keyring"
  keys               = ["foo"]
}

resource "google_kms_crypto_key_iam_binding" "crypto_key" {
  crypto_key_id = module.kms.keys["foo"]
  role          = "roles/cloudkms.cryptoKeyEncrypter"

  members = [
    local.user
  ]
}

resource "google_kms_key_ring_iam_binding" "key_ring" {
  key_ring_id = module.kms.keyring
  role        = "roles/editor"

  members = [
   local.user
  ]
}
```

**Comment by kpocius** [comment]
I'm setting up the permissions using the module itself, not separate resources. So just using your code as an example, it's more along the lines of:
```
module "kms" {
  source  = "terraform-google-modules/kms/google"
  version = "2.0.1" #change to 2.1.0

  project_id         = local.project_id
  location           = "europe"
  keyring            = "sample-keyring"
  keys               = ["foo"]
  set_encrypters_for = ["foo"]
  set_decrypters_for = ["foo"]
  encrypters         = [local.user]
  decrypters         = [local.user]
}
```

**Comment by bharathkkb** [comment]
Thanks for clarifying but I am still unable tor repro.

```tf
module "kms" {
  source  = "terraform-google-modules/kms/google"
  version = "2.0.1" #change to 2.1.0

  project_id         = local.project_id
  location           = "europe"
  keyring            = "sample-keyring"
  keys               = ["foo"]
  set_encrypters_for = ["foo"]
  set_decrypters_for = ["foo"]
  encrypters         = [local.user]
  decrypters         = [local.user]
}
```
after changing and re applying gave me 
```

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no
changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```
I also tried with provider versions explicitly pinned to `v3.90.1` and `v3.90.1` for `2.0.1` `v4.6.0` for `2.1.0` but both gave me no changes with TF `v1.1.2`

**Comment by kpocius** [comment]
Sorry for wasting your time. As it turns out, I can't reproduce this with a new key(ring) either, although my original problem still exists. But I now think that it's a local issue with the state or something, probably due to the fact that those keys that show a change were incorrectly imported somehow.
